tensor2tensor/data_generators/allen_downloader.py

```diff
@@ -0,0 +1,484 @@
+# coding=utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data download utilities.
+
+Usage: python download.py --tmp_dir <temporary directory> ...
+
+"""
+
+import math
+import os
+import shutil
+import tempfile
+
+import numpy as np
+import tensorflow as tf
+
+
+# Gracefully inform users of the need to install problem-specific deps.
+try:
+  from allensdk.api.queries.rma_api import RmaApi
+  from allensdk.api.queries.image_download_api import ImageDownloadApi
+  from PIL import Image
+  import pandas as pd
+except ImportError as e:
+  tf.logging.error("This problem requires the installation of additional "
+                   "dependencies. To proceed, please run "
+                   "`pip install tensor2tensor[allen]`.")
+  # Don't throw the error, it just buries the informative log message.
+  exit(1)
+
+
+flags = tf.flags
```

Comment:
no need for additional flags if it's all part of t2t-datagen, i.e. the problem needs to implement the `generate_data` method.